fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java

```diff
@@ -176,22 +234,23 @@ public void invalidatePermission(String username) {
       exception.set(e.getCause() != null ? e.getCause() : e);
     }
 
-    if (!successfulLookup.get()) {
-      log.error(
-          "Cannot get whole user permission for user {}, reason: {}",
-          username,
-          exception.get().getMessage()
-      );
-    }
-
     Id id = getPermissionCounterId
         .withTag("cached", cacheHit.get())
         .withTag("success", successfulLookup.get());
 
     if (!successfulLookup.get()) {
+      log.error(
+              "Cannot get whole user permission for user {}, reason: {}",
+              username,
+              exception.get().getMessage()
+      );
       id = id.withTag("legacyFallback", legacyFallback.get());
     }
 
+    if (legacyFallback.get()) {
+      permissionsCache.invalidate(username);
```

Comment:
this is new - if we returned a legacyFallback, invalidate the entry in the cache so we will try again next time we ask for the user